flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/saphana/SapHanaSqlStatementBuilder.java

```diff
@@ -44,22 +40,32 @@ protected String cleanToken(String token) {
 
     @Override
     protected Delimiter changeDelimiterIfNecessary(String line, Delimiter delimiter) {
-        if (StringUtils.countOccurrencesOf(statementStart, " ") < 4) {
-            statementStart += line;
-            statementStart += " ";
+
+        // TODO: When a line contains a CREATE X and and END statement, then the order is significant.
+        // TODO: There are a couple more statements allowing for nested begin/end blocks
+        if (line.startsWith("CREATE PROCEDURE") || line.startsWith("CREATE TRIGGER")) {
```

Comment:
Why is this nesting necessary? Can you point to a valid example SQL or better: add a test?